### ISSUE TITLE
[REVIEW]: DYCO: A Python package to dynamically detect and compensate for time lags in ecosystem time series

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2575]
**Submitting author:** @holukas (<a href="http://orcid.org/0000-0002-5569-0761">Lukas Hörtnagl</a>)
**Repository:** <a href="https://github.com/holukas/dyco" target ="_blank">https://github.com/holukas/dyco</a>
**Version:** v1.1.2
**Editor:** @KristinaRiemer
**Reviewer:** @mccabete, @r-barnes
**Archive:** <a href="https://doi.org/10.5281/zenodo.4964068" target="_blank">10.5281/zenodo.4964068</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/029f5d2be79462dd069e4541e7d16af2/status.svg)](https://joss.theoj.org/papers/029f5d2be79462dd069e4541e7d16af2)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/029f5d2be79462dd069e4541e7d16af2"><img src="https://joss.theoj.org/papers/029f5d2be79462dd069e4541e7d16af2/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/029f5d2be79462dd069e4541e7d16af2/status.svg)](https://joss.theoj.org/papers/029f5d2be79462dd069e4541e7d16af2)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/holukas/dyco" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@mccabete & @r-barnes, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @KristinaRiemer know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/holukas/dyco">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@holukas) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/holukas/dyco">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@holukas) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @mccabete, @r-barnes it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02575/joss.02575/10.21105.joss.02575.pdf)

**Comment by holukas** [comment]
Proof looks good!

**Comment by mccabete** [comment]
@KristinaRiemer I think the dyco repository aka [ target repository](https://gitlab.ethz.ch/holukas/dyco-dynamic-lag-compensation) is being hosted through ETH Zurich. I seem to need an ETH Zurich log-in to add issues. How do you want me to proceed?

**Comment by holukas** [comment]
@mccabete @KristinaRiemer I will check with ETH Zurich how they handle this and if they can open the registration.

**Comment by holukas** [comment]
@mccabete @KristinaRiemer I just got feedback from ETH. While the code itself is open to everyone, issues can only be added after registering an account with an email address from the ETH domain. However, there is the possibility to create a guest account for the purpose of adding issues. Would this still be in line with JOSS requirements?

**Comment by mccabete** [comment]
I can't speak to JOSS requirements, but I am happy to create a guest account for the review.

**Comment by KristinaRiemer** [comment]
@holukas if it's possible to use guest accounts to create issues in the repository, and the issues and their comments are able to be seen publicly, I think that would be the best route. Let me know if that ends up not being possible though!

**Comment by mccabete** [comment]
@holukas How should I request guest access?

**Comment by holukas** [comment]
@mccabete I am currently in contact with ETH to open a guest account and get back to you soon.

**Comment by KristinaRiemer** [comment]
Any progress on the guest accounts, @holukas?

**Comment by holukas** [comment]
@KristinaRiemer Sorry to keep you waiting, there were some surprising obstacles to overcome that should be solved now. @mccabete Can you get in contact with my via email (_lukas.hoertnagl [at] usys.ethz.ch_) so we can set up the guest account?

**Comment by r-barnes** [comment]
Just checking in to see if I should start on my review.

**Comment by mccabete** [comment]
Sent an email to you @holukas

**Comment by holukas** [comment]
Hi @r-barnes! Can you get in contact with me via email (lukas.hoertnagl [at] usys.ethz.ch) so we can set up the guest account for the review?

**Comment by holukas** [comment]
Sent credentials for guest account to @mccabete

**Comment by KristinaRiemer** [comment]
Hi @r-barnes & @mccabete. Have you been able to create and use your guest accounts? I only see issues from @holukas in the repository.

**Comment by mccabete** [comment]
Sorry about the delay! 

I am still running into issues - @holukas convinced ETH to give me a guest account (thanks!). However, it seems like ETH gave me an account that may/might not include github.ethz.ch privileges. I sent out an email to ETH Zurich's service desk today.

**Comment by mccabete** [comment]
I am essentially sitting on the review. @KristinaRiemer if the ETH service desk doesn't get back to me soon, I am willing to just post issues/comments on this thread instead. Is that joss-acceptable?

**Comment by KristinaRiemer** [comment]
@mccabete (and @r-barnes), I agree with your assessment. I think you should not worry about creating issues in the repo and just post your review comments and discussion here in this thread, so as to not slow down the review any more.

**Comment by r-barnes** [comment]
I worry somewhat about issues getting lost, but can do. @holukas - will that work for you?

**Comment by holukas** [comment]
Dear @KristinaRiemer @r-barnes @mccabete, after some back-and-forth I decided to move the code over to GitHub to make collaborations and the review work easier. I agree with @r-barnes that submitting issues here might get lost and is not ideal. @KristinaRiemer I would therefore like to cancel this current submission. I will then re-submit the same code, but this time using the GitHub repo. I hope the reviewers are still on-board to review the new submission. I am sorry for the inconvenience and hope for a smooth GitHub workflow. I already uploaded the code to https://github.com/holukas/dyco and I will make an official re-submission after this current one is closed. Thanks!

**Comment by KristinaRiemer** [comment]
That seems like a good solution @holukas! There's no need to reopen a new submission, I just changed the repo URL in the first comment of this review issue so we can continue using this issue.

**Comment by KristinaRiemer** [comment]
@mccabete & @r-barnes just checking in to make sure that the new GitHub repo's issues are accessible to y'all now, and to see if there are any questions or issues currently?

**Comment by KristinaRiemer** [comment]
Hi all! Hopefully everyone had a good holiday season. @r-barnes & @mccabete I'm checking about progress on your reviews of this software. Now that the issues permission problem has been taken care of by migrating everything over to GitHub, is there anything blocking your work or anything I can do to help move this forward?

**Comment by mccabete** [comment]
Thanks @KristinaRiemer  for the reminders. Sorry for not getting back to you both! @holukas, I added my comments as issues to the new repo. 

The Wiki link is now behind an ETH login, so there was some content I had access to that I no longer have access to. That said, I had already gone through the wiki content before the migration, so I was able to post my comments. Likely 98% of my review is now posted, and once the link is fixed I can verify the final remaining check marks. 

Thank you for being helpful and patient @KristinaRiemer  and @holukas

**Comment by r-barnes** [comment]
@KristinaRiemer : I can't seem to mark things off in my designated reviewer checklist above. Is there anything I can do about that? If not, I can keep track manually.

**Comment by KristinaRiemer** [comment]
@r-barnes let's see if this fixes that.

**Comment by KristinaRiemer** [comment]
@whedon re-invite @r-barnes as reviewer

**Comment by whedon** [comment]
The reviewer already has a pending invite.

@r-barnes please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by KristinaRiemer** [comment]
Ah, @r-barnes try going to that link and accepting the invite to see if that makes a difference?

**Comment by r-barnes** [comment]
I'm sorry, the link expired. Can we try again? I have my JOSS reviews at the top of the queue now, so I'll be ready for it.

**Comment by holukas** [comment]
@mccabete Thanks for the feedback. Unfortunately the Wiki got lost during the migration from GitLab to GitHub, so this time I will add it to GitHub directly. I will add an updated Wiki directly to GitHub in the next days.

**Comment by KristinaRiemer** [comment]
@whedon re-invite @r-barnes as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@r-barnes please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by r-barnes** [comment]
I've finished my review, per the above.

I think the biggest issue is that I'm left confused about whether Dyco is a tool I use kind of from the command-line or a library I use from within Python.

It looks as though Dyco produces a number of output directories and files, even if it's run from within Python. I think that's weird. Running commands from within a programming language should (mostly) give me products I can work with from within that language.

Conversely, the example I saw shows Dyco being run as a single, very large command from within Python to produce all of those outputs. So it's almost as though I have to open Python to run what's effectively a command-line command. If this is what's happening, then the authors should embrace the command-line nature of the work and modify their `setup.py` to add CLI commands.

Other than that, things look pretty good. The code appears clean and decently structured, though there is some missing documentation and dead code. Cleaning that up shouldn't be too difficult, though.

**Comment by KristinaRiemer** [comment]
Thank you @mccabete and @r-barnes for adding specific comments as issues to the software's repo, hopefully @holukas will have a chance to address all of them soon.

**Comment by holukas** [comment]
Thanks @mccabete and @r-barnes for the helpful comments! I will address the issues over the next days.

**Comment by KristinaRiemer** [comment]
Hi @holukas, just checking in to see how working on these issues is going for you?

**Comment by holukas** [comment]
Hi @KristinaRiemer, I am nearly finished with the Issues. I was not sure if I should close the issues if I think they are done, or if it is up to the reviewers to confirm that an issue was solved and then I can close them. What would be the correct procedure?

**Comment by KristinaRiemer** [comment]
@holukas I think we should do the latter. When you're done addressing all of the issues, let the reviewers know here. They can then review the work you've done and you can close the issues once they indicate they're satisfied with it. Hopefully that's the easiest way for @r-barnes and @mccabete to keep track of everything!

**Comment by holukas** [comment]
Dear @KristinaRiemer @r-barnes @mccabete,

thank you all for your input and feedback, highly appreciated! I have now addressed all issues and followed up on your suggestions. The script and documentation underwent quite some changes and I hope DYCO now meets the criteria for publication. Please let me know in case there are further changes you would like to see.

Cheers
Lukas

**Comment by KristinaRiemer** [comment]
Hi @r-barnes and @mccabete, have you had a chance yet to review the addressing of your issues in the [DYCO repo](https://github.com/holukas/dyco) yet? Please let me know if you need any help.

**Comment by KristinaRiemer** [comment]
Hi @r-barnes and @mccabete, this is just a reminder to look over your issues in the repo. If @holukas has sufficiently addressed them, feel free to comment and close. Hopefully this can be taken care of in the next week?

**Comment by r-barnes** [comment]
Sorry for going MIA. I've noticed github's notification system has improved, so hopefully I can make better use of that in the future. I'll take another look and see if we can get this wrapped up.

**Comment by KristinaRiemer** [comment]
Thanks for the update @r-barnes. I've struggled some with GitHub notifications, including accidentally turning off notifications for this entire repository (and therefore notifications for all submissions I'm editor for!), so I definitely understand that pain point.

**Comment by KristinaRiemer** [comment]
Hi @r-barnes and @mccabete, have you had a chance to look through the issues that @holukas has addressed?

**Comment by r-barnes** [comment]
I hope to have time this weekend or late Friday night :-(

**Comment by r-barnes** [comment]
@mccabete : I accidentally clicked some of your check boxes :-( There's not an undo button, so I tried to reverse as best I could, but may have made a couple of mistakes.

**Comment by r-barnes** [comment]
I feel like my concerns have been reasonably addressed and that the remaining issues I've left open shouldn't preclude moving this toward publication. My ability to really deep-dive into the results/processing of the package is hampered by my lack of domain knowledge here, but I think there's sufficient explanatory information that a user with this knowledge would be able to figure out what's going on and derive value from this software. I'm happy to approve this.

**Comment by KristinaRiemer** [comment]
Thanks for the review @r-barnes! @mccabete, it looks like you have a few items left on the review checklist and have two issues open in the software repo, so let me know if you need any help addressing those!

**Comment by arfon** [comment]
👋 folks, what's the status of this submission? It looks like we might be waiting on @mccabete to do a final check on the modifications by the author here?

**Comment by mccabete** [comment]
Thank you for the kind reminders @arfon  and @KristinaRiemer. Sorry, my github notifications we not reaching me. Thanks for your patience, @holukas. I just checked everything off!

**Comment by KristinaRiemer** [comment]
@arfon thank you for checking in on this! @mccabete it looks like the "functionality" and "summary" items in the check list are currently unchecked, do you have further comments about those areas?

**Comment by mccabete** [comment]
Nope! Sorry, just missed those
________________________________

**Comment by KristinaRiemer** [comment]
Okay, thanks @mccabete. I checked those off just for the sake of completion. 

@holukas there are a few more steps to go before publication. I'm going to check over the paper for typos and such, and it's suggested that you do also. Once that's done, I'll ask for archive links.

**Comment by KristinaRiemer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02575/joss.02575/10.21105.joss.02575.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02575/joss.02575/10.21105.joss.02575.pdf) :page_facing_up: :point_left:

**Comment by KristinaRiemer** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.2307/1941631 is OK
- 10.5194/amt-8-4197-2015 is OK
- 10.1016/S0022-1694(96)03194-0 is OK
- 10.1111/gcb.12518 is OK
- 10.1007/BF00128405 is OK
- 10.1515/intag-2017-0042 is OK
- 10.1111/gcb.14079 is OK
- 10.1007/978-94-007-2351-1 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by KristinaRiemer** [comment]
Okay, I found only a few small errors for you to fix in the paper @holukas: 

- Choose either bullets or numerals for the list at the top of the second page
- The first sentence in line 51 appears to be incomplete? I'm not sure if it should be joined to the following sentence or if there's something missing from it
- For the two figures, they only need one figure caption each. You can remove the one that's included when inserting each image if you want
- In line 124, the word "approx." should be fully spelled out
- In the last reference, in International Agrophysics, there are a few authors whose names should be fully included instead of condensed as an ellipsis

It also looks like the subscripts on CO2 and N20 aren't rendering. You put them correctly in the markdown, so I'm not sure what's going on and someone is looking into why that's happening.

**Comment by KristinaRiemer** [comment]
Hi @holukas, it looks like the syntax for the subscripts should be with `~` instead of `<sub>` for some reason. So `CO<sub>2</sub>` should instead be `CO~2~`. If you can modify those in the paper, we'll see if that works!

**Comment by holukas** [comment]
Thanks @KristinaRiemer for checking the PDF. I made the corrections and adjustments, except regarding the reference International Agrophysics. It seems like that the ellipsis was added when the PDF was generated and I am not sure I can change this from my side. In the .bib file all authors are included with their full name. Regarding the double captions for the figures, I now left the captions empty when importing the images, I hope that works.

**Comment by KristinaRiemer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02575/joss.02575/10.21105.joss.02575.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02575/joss.02575/10.21105.joss.02575.pdf) :page_facing_up: :point_left:

**Comment by KristinaRiemer** [comment]
Okay, I can't figure out why it truncated the author list for that reference, but I guess it's a default for a reason, so we can leave it. Thanks for making all those changes to the paper! 

The next steps for you to do @holukas are: 1) Create a tagged release of the GitHub repo (instructions [here](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release)) and post the version number here, and 2) generate a DOI by archiving the code on Zenodo, figshare, an institutional repository, etc. and provide that DOI. Make sure the archive has the same title as the paper title, and the same authors.

**Comment by danielskatz** [comment]
👋 @holukas - note that this is basically ready to accept and publish, and is waiting on you to do the 2 items above

**Comment by holukas** [comment]
@KristinaRiemer
1) I have created a new release on GitHub, the version number is v1.1.2: https://github.com/holukas/dyco/releases/tag/v1.1.2
2) I have archived the this version 1.1.2 on Zenodo, DOI: 10.5281/zenodo.4964068 
https://zenodo.org/record/4964068

**Comment by holukas** [comment]
With the submission coming to a close, I wanted to thank @mccabete @r-barnes for reviewing the code. I highly appreciate your helpful comments and suggestions!

**Comment by KristinaRiemer** [comment]
@whedon set v1.1.2 as version

**Comment by whedon** [comment]
OK. v1.1.2 is the version.

**Comment by KristinaRiemer** [comment]
@holukas thanks for getting these together! 

For the Zenodo archive, could you change the name of the archive to the name of the paper ("DYCO: A Python package to dynamically detect and1compensate for time lags in ecosystem time series")? Additionally, could you make the author name your name as in the paper?

**Comment by holukas** [comment]
@KristinaRiemer I have updated the Zenodo record accordingly: https://zenodo.org/record/4964068

**Comment by KristinaRiemer** [comment]
@whedon set 10.5281/zenodo.4964068 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4964068" target="_blank">10.5281/zenodo.4964068</a> is the archive.

**Comment by KristinaRiemer** [comment]
Excellent! That looks perfect. 

I'm now turning this over to the EiCs for final acceptance and publication.

**Comment by KristinaRiemer** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.2307/1941631 is OK
- 10.5194/amt-8-4197-2015 is OK
- 10.1016/S0022-1694(96)03194-0 is OK
- 10.1111/gcb.12518 is OK
- 10.1007/BF00128405 is OK
- 10.1515/intag-2017-0042 is OK
- 10.1111/gcb.14079 is OK
- 10.1007/978-94-007-2351-1 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2395

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2395, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@holukas - As part of a final proofread, I have suggested some changes in https://github.com/holukas/dyco/pull/19 - please merge these or let me know what you disagree with, then we can proceed with the publishing.

**Comment by holukas** [comment]
@danielskatz There is only one small thing that needs to be changed, I commented directly in https://github.com/holukas/dyco/pull/19

**Comment by danielskatz** [comment]
Sorry, I don't see any comment in https://github.com/holukas/dyco/pull/19

**Comment by holukas** [comment]
@danielskatz Should be visible now (I forgot to hit the button it seems...).

**Comment by danielskatz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02575/joss.02575/10.21105.joss.02575.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02575/joss.02575/10.21105.joss.02575.pdf) :page_facing_up: :point_left:

**Comment by danielskatz** [comment]
@holukas - please confirm this looks good now, and I'll go ahead and publish it

**Comment by holukas** [comment]
@danielskatz There is only one small typo that I found. In the PDF L131 please change "compensated" to "compensates", rest looks good.

**Comment by danielskatz** [comment]
You can and should do that, since it's in your repo :)  When you do it, run ```@whedon generate pdf``` again and then let me know that it's ready to publish

**Comment by holukas** [comment]
Yes you are right :)

**Comment by holukas** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02575/joss.02575/10.21105.joss.02575.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02575/joss.02575/10.21105.joss.02575.pdf) :page_facing_up: :point_left:

**Comment by holukas** [comment]
Looks good and is ready to publish!

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1405271915002929154) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2396
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02575](https://doi.org/10.21105/joss.02575)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to @holukas (Lukas Hörtnagl)!!

And thanks to @mccabete and @r-barnes for reviewing, and @KristinaRiemer for editing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02575/status.svg)](https://doi.org/10.21105/joss.02575)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02575">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02575/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02575/status.svg
   :target: https://doi.org/10.21105/joss.02575
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02575/status.svg)](https://doi.org/10.21105/joss.02575)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by holukas** [comment]
Thanks a lot for your support @KristinaRiemer @mccabete @r-barnes @danielskatz (and whedon)!

**Comment by KristinaRiemer** [comment]
Congrats @holukas, and thanks for reviewing @mccabete and @r-barnes!
